src/relations/postgresql_client.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2022 Canonical Ltd.
```

Comment:
I think this file should be changed to `postgresql_provider.py` since this is provider code. (Also mongodb follows this schema and it would be nice if they followed the same pattern.)